data/convert.go

```diff
@@ -53,7 +53,10 @@ func NewWith(convert StructOptions, value interface{}) Value {
 	case reflect.String:
 		return String(v.String())
 	case reflect.Slice:
-		var slice []Value
+		if v.IsNil() {
+			return Null{}
```

Comment:
Thats the behavior that we want it in our system, but I wasn't sure if we wanted to push that on consumers of this library.

Previous behavior:
```
[]string => nil
[]string{} => nil
```

The new behavior:
```
[]string => nil
[]string{} => []
```

It seems this matches expectations better, if you have a need to differentiate the two (not present vs emtpy).